File: 2025-05_Spring-101-Java-Maven/src/main/java/com/happn/worshop/domain/UserService.java
package com.happn.worshop.domain;

import lombok.RequiredArgsConstructor;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.beans.factory.annotation.Qualifier;
import org.springframework.stereotype.Service;

import java.util.Optional;
import java.util.UUID;

@RequiredArgsConstructor
@Service
public class UserService {

    private final UserRepository userRepository;

    public Optional<User> findById(UUID id) {
        return userRepository.findById(id);
    }

    public User createUser(String name, String email) {
        return userRepository.save(User.builder()
                .id(UUID.randomUUID())
                .email(email)
                .name(name)
                .build());
    }


}

File: 2025-05_Spring-101-Kotlin-Gradle/src/main/kotlin/com/happn/workshop/domain/UserService.kt
package com.happn.workshop.domain

import org.springframework.beans.factory.annotation.Qualifier
import org.springframework.stereotype.Service
import java.util.UUID

@Service
class UserService(
    @Qualifier("userPostgresRepository")
    private val repo: UserRepository,
) {
    fun findById(id: UUID): User? = repo.findById(id)
    fun save(user: UserCreate) : String = repo.save(user)
}